HLTriggerOffline/Muon/src/HLTMuonPlotter.cc

```diff
@@ -106,171 +114,238 @@ void HLTMuonPlotter::beginRun(DQMStore::IBooker &iBooker, const Run &iRun, const
-    if (!recMuonSelector_)
-      recMuonSelector_ = new StringCutObjectSelector<reco::Muon>(recMuonCut_);
+    LogTrace("HLTMuonPlotter") << "[HLTMuonPlotter] source=" << source;
 
     // Make each good gen/rec muon into the base cand for a MatchStruct
-    vector<MatchStruct> matches;
-    if (source == "gen" && genParticles.isValid())
-      for (size_t i = 0; i < genParticles->size(); i++)
-        if ((*genMuonSelector_)(genParticles->at(i)))
-          matches.push_back(MatchStruct(&genParticles->at(i)));
-    if (source == "rec" && recMuons.isValid())
-      for (size_t i = 0; i < recMuons->size(); i++)
-        if ((*recMuonSelector_)(recMuons->at(i)))
-          matches.push_back(MatchStruct(&recMuons->at(i)));
+    std::vector<MatchStruct> matches;
+
+    if (source == "gen") {
+      if (genParticlesHandle.isValid()) {
+        matches.reserve(matches.size() + genParticlesHandle->size());
+        for (auto const &genp : *genParticlesHandle)
+          if (genMuonSelector_(genp))
+            matches.emplace_back(MatchStruct(&genp));
+      }
+    } else if (source == "rec") {
+      if (recoMuonsHandle.isValid()) {
+        matches.reserve(matches.size() + recoMuonsHandle->size());
+        for (auto const &recomu : *recoMuonsHandle)
+          if (recMuonSelector_(recomu))
+            matches.emplace_back(MatchStruct(&recomu));
+      }
+    }
 
     // Sort the MatchStructs by pT for later filling of turn-on curve
-    sort(matches.begin(), matches.end(), matchesByDescendingPt());
+    std::sort(matches.begin(), matches.end(), matchesByDescendingPt());
 
-    const bool isDoubleMuonPath = (hltPath_.find("Double") != string::npos);
-    const size_t nFilters = moduleLabels_.size();
-    const size_t nSteps = stepLabels_.size();
-    const size_t nStepsHlt = nSteps - 2;
-    const int nObjectsToPassPath = (isDoubleMuonPath) ? 2 : 1;
     l1t::MuonVectorRef candsL1;
-    vector<vector<RecoChargedCandidateRef>> refsHlt(nStepsHlt);
-    vector<vector<const RecoChargedCandidate *>> candsHlt(nStepsHlt);
-
-    for (size_t i = 0; i < nFilters; i++) {
-      const int hltStep = i - 1;
-      InputTag tag = InputTag(moduleLabels_[i], "", hltProcessName_);
-      size_t iFilter = rawTriggerEvent->filterIndex(tag);
-      if (iFilter < rawTriggerEvent->size()) {
-        if (i == 0)
-          rawTriggerEvent->getObjects(iFilter, TriggerL1Mu, candsL1);
-        else
-          rawTriggerEvent->getObjects(iFilter, TriggerMuon, refsHlt[hltStep]);
+    std::vector<std::vector<reco::RecoChargedCandidateRef>> refsHlt(nStepsHlt);
+    std::vector<std::vector<const reco::RecoChargedCandidate *>> candsHlt(nStepsHlt);
+
+    for (int idx = 0; idx < nFilters; ++idx) {
+      auto const moduleLabelStripped =
+          edm::path_configuration::removeSchedulingTokensFromModuleLabel(moduleLabels_[idx]);
+      auto const iTag = edm::InputTag(moduleLabelStripped, "", hltProcessName_);
```

Comment:
The module 'label' here (i.e. `moduleLabels_[idx]`) can contain special characters to denote how the module/filter was used in the configuration (e.g. `moduleLabels_[idx] == "-hltModule"` if the result of the filter was ignored, because used with `cms.ignore( hltModule )`). In order to find the corresponding trigger objects, the InputTag needs to be built with the 'stripped' version of the module label (meaning, without special characters).